src/main/java/ai/promoted/delivery/client/PromotedDeliveryClient.java

```diff
@@ -127,6 +133,14 @@ private PromotedDeliveryClient(String deliveryEndpoint, String deliveryApiKey,
    */
   public DeliveryResponse deliver(DeliveryRequest deliveryRequest) throws DeliveryException {
 
+    boolean isShadowTraffic = shouldSendShadowTraffic();
+    
+    if (performChecks) {
+      List<String> validationErrors = deliveryRequest.validate(isShadowTraffic);
+      for (String validationError : validationErrors) {
+        LOGGER.warning("Delivery Request Validation Error: " + validationError);
```

Comment:
Side comment - What's the approach for ~handleError in this client?  What if a timeout happens in production?

Reply:
It logs like this using standard Java logging, afaik the custom handleError in the other SDKs has never been used for anything other than logging in any event. Timeouts are the custom java.net.http.HttpClient behavior which is to throw (see ApiDelivery.java) which will kick the flow into the handler a few lines down.